lib/sentry/sourcecodeloader.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (C) 2019 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package sentry
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/getsentry/raven-go"
+	"github.com/syncthing/syncthing/lib/build"
+)
+
+const (
+	packagePrefix = "github.com/syncthing/syncthing"
+	urlPrefix     = "https://raw.githubusercontent.com/syncthing/syncthing/"
+)
+
+type chainedSourceCodeLoader struct {
+	loaders []raven.SourceCodeLoader
+}
+
+func (l *chainedSourceCodeLoader) Load(filename string, line, context int) ([][]byte, int) {
+	for _, loader := range l.loaders {
+		data, n := loader.Load(filename, line, context)
+		if data != nil {
+			return data, n
+		}
+	}
+	return nil, 0
+}
+
+type githubSourceCodeLoader struct {
```

Comment:
What is all this stuff ... ?

Reply:
I think we might not need this once we have the github integration up and running, but for now we implement a custom code loader which allows us to load code around the location of the crash, so that we'd get nice stack traces.

Reply:
It feels ... bizarre, that the code that is crashing should go looking for its own source on GitHub. This very much feels like something that should happen server side based on the backtrace...

Reply:
I wonder if we could package all source code as part of the build, and use that.

It seems the github integration doesn't do source code lookup from gitbub.

Reply:
Source code view is nice and all, but wouldn't the info in the normal trace (info on function/arguments and file/lineno) be enough?